File: backend/app/models/discipline.py
import enum


class Discipline(str, enum.Enum):
    CHEMISTRY = "chemistry"
    PHYSICS = "physics"
    BIOLOGY = "biology"


def subject_label(discipline: Discipline) -> str:
    return {
        Discipline.CHEMISTRY: "化学实验",
        Discipline.PHYSICS: "物理实验",
        Discipline.BIOLOGY: "生物实验",
    }[discipline]


def materials_label(discipline: Discipline) -> str:
    return {
        Discipline.CHEMISTRY: "试剂与仪器",
        Discipline.PHYSICS: "器材与材料",
        Discipline.BIOLOGY: "材料、试剂与仪器",
    }[discipline]

File: frontend/src/constants/discipline.ts
export type Discipline = 'chemistry' | 'physics' | 'biology'

export const DISCIPLINE_LABEL: Record<Discipline, string> = {
  chemistry: '化学',
  physics: '物理',
  biology: '生物',
}

export const MATERIALS_LABEL: Record<Discipline, string> = {
  chemistry: '试剂与仪器',
  physics: '器材与材料',
  biology: '材料、试剂与仪器',
}
